packages/fiber/src/core/stages.ts

```diff
@@ -0,0 +1,109 @@
+import { MutableRefObject } from 'react'
+import { StoreApi, UseBoundStore } from 'zustand'
+import { RootState } from './store'
+
+export interface UpdateCallback {
+  (state: RootState, delta: number, frame?: THREE.XRFrame | undefined): void
+}
+
+export type UpdateCallbackRef = MutableRefObject<UpdateCallback>
+type Store = UseBoundStore<RootState, StoreApi<RootState>>
+export type UpdateSubscription = { ref: UpdateCallbackRef; store: Store }
+
+export type FixedStageOptions = { fixedStep?: number; maxSubSteps?: number }
+
+export class Stage {
```

Comment:
I'd add a comment here explaining in detail what the class does and what contract to keep when extending - I understand this is very implicit in the ts but it's still useful to have it in plain english